### ISSUE TITLE
prevent redirects when not logged into moodle

### ISSUE THREAD

**Issue by kaseyhackspace** [catalyst/moodle-tool_redirects#17]
It would be good if the plugin prevents redirects if the user is not logged into moodle. The current behavior is that custom redirects are still applied even if you access the "redirect from" link in incognito where a user is not logged in.

**Comment by brendanheywood** [comment]
thanks @kaseyhackspace 

Can you give an example use case for why you'd want that? If you added a redirect using say apache or nginx you typically don't care about the user at all its just a blanket redirect.

**Comment by kaseyhackspace** [comment]
Hi @brendanheywood , thanks for responding. We're mass redirecting a lot of `/mod/page/view.php?id=<id>` URLs on one of our courses with where the student needs to be enrolled in to access the resource. The default behavior when you visit  `/mod/page/view.php?id=<id>` and you are not logged in is you are redirected to the moodle login page. We would still want this in place so students don't go sharing the links publicly :)

**Comment by brendanheywood** [comment]
Just had another client ask for something similar to this but the exact opposite.

I'm gonna propose we add a flag syntax which we can reuse for other features, vaguely inspired by the apache syntax:

```
#^/target#=>/destination
#^/target#=>/destination [anon]
#^/target#=>/destination [auth]
#^/target#=>/destination [auth=manual]
```

The alternative is we ditch specifying redirects as a text area and move to editing each one as a record with a form and these could be tick boxes.

**Comment by kaseyhackspace** [comment]
Hi @brendanheywood , I vote for the flag syntax on the text area. Reason is due to the number of redirects we have to make, I made a script that generates the redirect rules that I put in the text area. If it was a per record approach, might be harder to maintain hundreds of links by batch?

**Comment by brendanheywood** [comment]
Yeah textarea certainly makes that easier. We could expose it with a cli / web service to add and remove them pragmatically but its just more work

**Comment by kaseyhackspace** [comment]
Yeah the KISS principle applies to this one.
